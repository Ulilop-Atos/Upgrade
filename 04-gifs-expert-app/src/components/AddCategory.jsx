import PropTypes from 'prop-types';
import React, { useState } from 'react';

const AddCategory = ({ updateArray }) => {

  const [inputValue, setInputValue] = useState('');

  const handleInputChange = (event) => setInputValue(event.target.value); 

  const validate = (str) => {
    return str.match(/^[\S][\w\s]*\S$/gmiu);
  }

  const handleSubmit = (event) => {
    event.preventDefault();
    const str = event.target[0].value;
    if(updateArray && validate(str)) {
      // Callback edit
      updateArray(prev => [str, ...prev]);
    }
  };

  return ( 
    <form onSubmit={ handleSubmit }>
      <input 
        type='text'
        value={inputValue}
        onChange={handleInputChange}
        placeholder='Search'
        pattern='^[\S][\w\s]*\S$'
      />
    </form>
  );
};

AddCategory.propTypes = {
  updateArray: PropTypes.func.isRequired,
};

export default AddCategory;
