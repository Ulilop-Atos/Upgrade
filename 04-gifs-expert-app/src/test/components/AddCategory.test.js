import '@testing-library/jest-dom';
import { shallow } from "enzyme";
import AddCategory from "../../components/AddCategory";

describe('Tests for <AddCategory />', () => {

  let wrapper
  const setCategories = jest.fn();
  
  beforeEach( () => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test('should display correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should change the input box', () => {
    const value = 'Test';
    // The obj in the 2nd arg is the event
    wrapper.find('input').simulate('change', { target: { value: value } });
    expect(wrapper.find('p').text()).toBe(value);

  });

  test('should not post info on Submit', () => {
    wrapper.find('form').simulate('submit', { preventDefault(){} })
    expect(setCategories).not.toHaveBeenCalled()  
  });

  test('should call setCategories and clean the input box', () => { 
    const value = 'Test';
    // The obj in the 2nd arg is the event
    wrapper.find('input').simulate('change', { target: { value } });
    wrapper.find('form').simulate('submit', { preventDefault(){} });
    expect(setCategories).toHaveBeenCalledTimes(1);
    expect(setCategories).toHaveBeenCalledWith( expect.any(Function) );
    expect(wrapper.find('input').prop('value')).toBe('');
   })
});