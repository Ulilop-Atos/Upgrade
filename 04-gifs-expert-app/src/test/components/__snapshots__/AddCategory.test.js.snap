// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for <AddCategory /> should display correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    pattern="^[\\\\S][\\\\w\\\\s]*\\\\S$"
    placeholder="Search"
    type="text"
    value=""
  />
</form>
`;
